coral-component-list/src/scripts/SelectList.js

```diff
@@ -218,13 +218,35 @@ class SelectList extends BaseComponent(HTMLElement) {
   get _tabTarget() {
     return this.__tabTarget || null;
   }
+  
   set _tabTarget(value) {
     this.__tabTarget = value;
-  
-    // Set all but the current set _tabTarget to not be a tab target:
-    this.items.getAll().forEach((item) => {
-      item.setAttribute('tabindex', item === value ? 0 : -1);
-    });
+    if (this.groups && this._groups.getAll().length > 0) {
+      // set first item or selected item to be tab target in each group
+      var groups = this._groups;
+      groups.getAll().forEach((group) => {
+        if (group.items.getAll().length > 0) {
+          var firstChild = group.firstElementChild;
+          firstChild.setAttribute('tabindex', 0);
```

Comment:
@hameed87 I think this code to set firstItem tab-target, should go in function _resetTabTarget instead of this.
Could you please try?